tokio-udev/src/lib.rs

```diff
@@ -137,41 +136,34 @@ impl Stream for MonitorSocket {
 
     fn poll_next(
         self: Pin<&mut Self>,
-        cx: &mut std::task::Context,
+        ctx: &mut std::task::Context,
     ) -> Poll<Option<Self::Item>> {
-        self.inner.lock().unwrap().poll_receive(cx)
+        self.inner.lock().unwrap().poll_receive(ctx)
     }
 }
 
 struct Inner {
-    io: PollEvented<mio_udev::MonitorSocket>,
+    fd: AsyncFd<mio_udev::MonitorSocket>,
 }
 
 impl Inner {
     fn new(monitor: mio_udev::MonitorSocket) -> io::Result<Inner> {
         Ok(Inner {
-            io: PollEvented::new(monitor)?,
+            fd: AsyncFd::new(monitor)?,
         })
     }
 
     fn poll_receive(
         &mut self,
-        cx: &mut std::task::Context,
+        ctx: &mut std::task::Context,
     ) -> Poll<Option<mio_udev::Event>> {
-        if let Poll::Pending =
-            self.io.poll_read_ready(cx, mio::Ready::readable())
-        {
-            return Poll::Pending;
-        }
-
-        match self.io.get_mut().next() {
-            Some(event) => Poll::Ready(Some(event)),
-            None => {
-                self.io
-                    .clear_read_ready(cx, mio::Ready::readable())
-                    .unwrap();
-                Poll::Pending
+        match self.fd.poll_read_ready(ctx) {
+            Poll::Ready(Ok(mut ready_guard)) => {
+                ready_guard.clear_ready();
+                Poll::Ready(self.fd.get_mut().next())
             }
+            Poll::Ready(Err(_)) => Poll::Ready(None),
```

Comment:
Should we be returning `Poll::Ready(None)` for errors? Not sure here

Reply:
We can fix this by changing the `Item` type of the `Stream` implementation to be a `io::Result<mio_udev::Event>` as returning `None` for every error to signal the end of the stream might not be wanted and will hide the error, so yeah, go for it :-)

Reply:
Sounds good. Since this is technically an API break, would this involve any kind of semver change?

Reply:
We'll have to update the semver, but I can do it once this is merged to publish it directly to crates.io

Reply:
Fixed this to use the result type